actix-server/src/handlers/contact_handler.rs

```diff
@@ -149,3 +150,44 @@ pub async fn update_contact(
     .await?;
     Ok(HttpResponse::Ok().json(contact))
 }
+
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+pub struct ContactList {
+    pub contacts: Vec<Contact>,
+    pub total: i64,
+}
+
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+pub struct ListContactsQuery {
+    pub limit: Option<i64>,
+    pub offset: Option<i64>,
+}
+
+#[utoipa::path(
+  get,
+  path = "/contacts/list",
+  context_path = "/api",
+  tag = "contacts",
+  responses(
+      (status = 200, description = "JSON object representing the requested contact", body = ContactList),
+      (status = 401, description = "Service error relating to authentication status of the user", body = ErrorRespPayload),
+  ),
+  security(
+      ("ApiKey" = ["readonly"]),
+  ),
+  params(
+    ("limit" = i64, Query, description = "The number of contacts to return"),
+    ("offset" = i64, Query, description = "The offset to start from"),
```

Comment:
```suggestion
    ("offset" = uuid, Query, description = "The offset to start from"),
```

When you offset by a `number` it doesn't use the index and is really slow. Not the best blog, but - https://geronimovelasco.medium.com/how-limit-and-offset-works-behind-the-scenes-in-postgresql-a09c2c063405